src/Symfony/Bridge/PhpUnit/DeprecationErrorHandler/Configuration.php

```diff
@@ -125,6 +158,40 @@ public function tolerates(array $deprecationGroups)
         return true;
     }
 
+    /**
+     * @return bool
+     */
+    public function isBaselineDeprecation(Deprecation $deprecation)
+    {
+        $result = \in_array($deprecation->getMessage(), $this->baselineDeprecations, true);
+        if (!$result && $this->generateBaseline) {
+            $this->baselineDeprecations[] = $deprecation->getMessage();
+            $result = true;
+        }
+
+        return $result;
+    }
+
+    /**
+     * @return bool
+     */
+    public function isGeneratingBaseline()
+    {
+        return $this->generateBaseline;
+    }
+
+    public function getBaselineFile()
+    {
+        return $this->baselineFile;
+    }
+
+    public function writeBaseline()
+    {
+        if (false === @file_put_contents($this->baselineFile, json_encode($this->baselineDeprecations))) {
```

Comment:
I'd suggest using file_put_contents and no wrapper. This will display a more accurate error and should be rare anyway.

Reply:
`\JSON_PRETTY_PRINT | \JSON_UNESCAPED_SLASHES` so that the file can be committed and diffs remain meaningful.

Reply:
Having the wrapper makes testing a bit easier.